go/kbfs/libkbfs/settings_db.go

```diff
@@ -0,0 +1,142 @@
+package libkbfs
+
+import (
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/keybase/client/go/kbfs/idutil"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/pkg/errors"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/storage"
+	"golang.org/x/net/context"
+)
+
+const (
+	// Where in config.StorageRoot() we store settings information.
+	settingsDBDir           = "kbfs_settings"
+	settingsDBVersionString = "v1"
+	settingsDBName          = "kbfsSettings.leveldb"
+
+	// Settings keys
+	notificationThresholdKey = "notificationThreshold"
+)
+
+// ErrNoSettingsDB is returned when there is no settings DB potentially due to
+// multiple concurrent KBFS instances.
+var ErrNoSettingsDB = errors.New("no settings DB")
+
+var errNoSession = errors.New("no session")
+
+// SettingsDB stores KBFS user settings for a given device.
+type SettingsDB struct {
+	*LevelDb
+	sessionGetter idutil.CurrentSessionGetter
+
+	cache map[string][]byte
```

Comment:
Why is there a cache here? Is this DB accessed often enough for it to be necessary?

Reply:
I anticipate it being accessed on every disk cache access. But levelDB might already have a cache.